src/js/date.js

```diff
@@ -104,51 +107,37 @@ export const getLocaleDayMonthYear = (locale = 'en-uk', date = new Date()) => {
 };
 
 export const isDateValid = (locale = 'en-uk', inputValue = '') => {
-  let splittedInput = [];
-  let orderedInput = [];
-  let date = null;
-
-  if (locale === 'de-ch') {
-    splittedInput = inputValue.split('.');
-    orderedInput = [splittedInput[1], splittedInput[0], splittedInput[2]];
-    date = new Date(orderedInput.join('.'));
-  } else {
-    date = new Date(inputValue);
-  }
+  // year, monthIndex, day
+  const blueprint = new Date(2017, 2, 23);
 
-  let objDate = '';
-  const numeric = 'numeric';
-  const twoDigit = '2-digit';
-  const selectors = [numeric, twoDigit];
-  const filters = ['year', 'month', 'day'];
-  let options = {};
-  let valid = false;
+  // find out which out of 4 valid seperator the current locale has
+  const localisedBlueprintDate = new Intl.DateTimeFormat(locale).format(blueprint);
```

Comment:
We can query supported locales with:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/DateTimeFormat/supportedLocalesOf

Reply:
added local check that throws an expetion. thanks